crates/plex-api/src/server/mod.rs

```diff
@@ -174,6 +200,43 @@ impl Server {
         }
     }
 
+    /// Marks a media item as fully watched and increases its view count.
+    pub async fn watched<M: MediaItem>(&self, item: &M) -> Result {
+        let path = format!(
+            "{SERVER_SCROBBLE}?identifier=com.plexapp.plugins.library&key={}",
+            item.rating_key()
+        );
+
+        self.client.get(path).consume().await
+    }
+
+    /// Marks a media item as unwatched.
+    pub async fn unwatched<M: MediaItem>(&self, item: &M) -> Result {
+        let path = format!(
+            "{SERVER_UNSCROBBLE}?identifier=com.plexapp.plugins.library&key={}",
+            item.rating_key()
+        );
+
+        self.client.get(path).consume().await
+    }
+
+    /// Sets a media items playback position in milliseconds. The server currently ignores any
+    /// positions equal to or less than 60000ms
+    pub async fn update_timeline<M: MediaItem>(&self, item: &M, position: u64) -> Result {
+        let rating_key = item.rating_key();
+        let query = Query::new()
+            .param("key", format!("/library/metadata/{rating_key}"))
+            .param("ratingKey", rating_key)
+            .param("offline", "1")
+            .param("state", "stopped")
```

Comment:
What are those? :) Can we have those values as arguments?

Reply:
We could pass the state param, it appears to accept "paused", "buffering", "stopped" and "playing". But it's not clear to me what effect those actually have, I didn't see any obvious difference to how the server behaves.

The offline param is what gets sent by mobile clients when syncing playback that happened while offline. It seemed that without this there is extra work to do, you have to create a playback queue to tell the server you are playing an item, then updated the timeline.

Reply:
Why does it always have to be so complicated with Plex 🤦‍♂️ 

Ok, let's start with the current implementation. We'll always have a chance to rewrite it later.